src/test/java/pl/touk/parking/InMemoryParkingUsageRepository.java

```diff
@@ -0,0 +1,43 @@
+package pl.touk.parking;
+
+import pl.touk.parking.domain.ParkingUsage;
+import pl.touk.parking.repository.ParkingUsageRepository;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class InMemoryParkingUsageRepository implements ParkingUsageRepository {
+
+    private final IdGenerator idGenerator = new IdGenerator();
+
+    private final Map<Long, ParkingUsage> usages = new HashMap<>();
+
+    @Override
+    public void save(ParkingUsage usage) {
+        if (usage.getId() == null)
+            usage.setId(idGenerator.nextId());
+        usages.put(usage.getId(), usage);
+    }
+
+    @Override
+    public Optional<ParkingUsage> findByDriverVehicleNumberAndEndTimeIsNull(String vehicleNumber) {
+        for (ParkingUsage usage : usages.values()) {
```

Comment:
czemu nie na streamach?

Reply:
```java
        return usages.values().stream()
                .filter(usage -> !usage.hasEnded() && usage.isByDriverOfVehicleWith(vehicleNumber))
                .findFirst();
```
Nie wiem czemu, chyba przez stare nawyki...

Reply:
Pamiętaj, że jeżeli nie zależy ci na kolejności, to lepiej stosować `findAny()` (patrz JavaDoc)